src/Native/Unix/System.Security.Cryptography.Native/pal_bio.cpp

```diff
@@ -52,3 +118,29 @@ extern "C" int32_t CryptoNative_BioCtrlPending(BIO* bio)
     assert(result <= INT32_MAX);
     return static_cast<int32_t>(result);
 }
+
+extern "C" void CryptoNative_BioSetAppData(BIO* bio, void* data)
+{
+    BIO_set_ex_data(bio, 0, data);
+}
+
+extern "C" void CryptoNative_BioSetWriteFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_WRITE);
+}
+
+extern "C" void CryptoNative_BioSetShoudRetryReadFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY | BIO_FLAGS_READ);
+}
```

Comment:
Rather than exposing two BIO_set_flags from the shim, should we just expose one that takes the flags to set?  I don't have a strong opinion, but in general where possible we've tried to keep the shim methods as close to the original as possible.